generator/internal/language/templates/rust/crate/src/transport.rs.mustache

```diff
@@ -56,10 +56,10 @@ impl crate::traits::{{NameToPascal}} for {{NameToPascal}} {
     ) -> Result<{{OutputTypeName}}> {
         let options = options.set_default_idempotency(reqwest::Method::{{HTTPMethod}}.is_idempotent());
         let builder = self.inner.builder(
-            reqwest::Method::{{HTTPMethod}}, format!("{{HTTPPathFmt}}"
-               {{#HTTPPathArgs}}
-               , {{{.}}}
-               {{/HTTPPathArgs}}
+            reqwest::Method::{{HTTPMethod}},
+            format!("{{#HTTPPathInfo.PathTemplate}}{{#Literal}}/{{Value}}{{/Literal}}{{#FieldPath}}/{}{{/FieldPath}}{{#Verb}}:{{Value}}{{/Verb}}{{/HTTPPathInfo.PathTemplate}}"{{#HTTPPathInfo.PathTemplate}}{{#FieldPath}}
```

Comment:
I have to say, I am not sure if having the code here is the template is any better than having it in the codec... At least  I could write unit tests for the codec.

I like the rest of the changes, but this is hard to read.  Let's go with this and maybe experiment with moving all of this to the Codec on a separate PR?

Reply:
I think this could be reimplemented as a Codec in the HTTPPathInfo. I was trying to keep the `HTTPPathFmt` and `HTTPPathArgs` logic side by side, as they depend on each other, but I agree that building the rust string template in mustache reduces readability.

Reply:
I've updated this code to take the fully formed format string from the codec, and added helper methods to the codec so the template doesn't have to iterate through irrelevant segments, it got a bit cleaner.